### PR TITLE
Removed usage of Guava

### PR DESCRIPTION
**Pull request by rsandell**
To secure a future upgrade in core.

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java [hunk 1]
```diff
@@ -129,7 +131,7 @@ public void sendInput(byte[] input) {
 
     public boolean slurpOutput(FastPipedOutputStream stdout, FastPipedOutputStream stderr) throws IOException {
         LOGGER.log(Level.FINE, () -> "--> SlurpOutput");
-        ImmutableMap<String, FastPipedOutputStream> streams = ImmutableMap.of("stdout", stdout, "stderr", stderr);
+        Map<String, FastPipedOutputStream> streams = new HashMap<>(); streams.put("stdout", stdout); streams.put("stderr", stderr);
```
Comment (MRamonLeon) [inline src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java:134]: It's not an immutable but I think it's fine. We only use it internally. Maybe you can initialize to 2 elements, as they're the only elements we will use

File: src/main/java/hudson/plugins/ec2/win/winrm/request/OpenShellRequest.java [hunk 2]
```diff
@@ -21,7 +20,7 @@ public OpenShellRequest(URL url) {
 
     protected void construct() {
         try {
-            defaultHeader().action(new URI("http://schemas.xmlsoap.org/ws/2004/09/transfer/Create")).resourceURI(new URI("http://schemas.microsoft.com/wbem/wsman/1/windows/shell/cmd")).options(ImmutableList.of(new Option("WINRS_NOPROFILE", "FALSE"), new Option("WINRS_CODEPAGE", "437")));
+            defaultHeader().action(new URI("http://schemas.xmlsoap.org/ws/2004/09/transfer/Create")).resourceURI(new URI("http://schemas.microsoft.com/wbem/wsman/1/windows/shell/cmd")).options(Arrays.asList(new Option("WINRS_NOPROFILE", "FALSE"), new Option("WINRS_CODEPAGE", "437")));
```
Comment (MRamonLeon) [inline src/main/java/hudson/plugins/ec2/win/winrm/request/OpenShellRequest.java:23]: It's not immutable as before but we don't use the list afterwards anyway.

File: src/main/java/hudson/plugins/ec2/win/winrm/soap/Header.java [hunk 3]
```diff
@@ -15,9 +17,9 @@
     private final  String action;
     private final  String shellId;
     private final  String resourceURI;
-    private final  ImmutableList<Option> optionSet;
+    private final List<Option> optionSet;
 
-    Header(String to, String replyTo, String maxEnvelopeSize, String timeout, String locale, String id, String action, String shellId, String resourceURI, ImmutableList<Option> optionSet) {
+    Header(String to, String replyTo, String maxEnvelopeSize, String timeout, String locale, String id, String action, String shellId, String resourceURI, List<Option> optionSet) {
```
Comment (MRamonLeon) [inline src/main/java/hudson/plugins/ec2/win/winrm/soap/Header.java:22]: Better API

File: src/main/java/hudson/plugins/ec2/win/winrm/soap/Header.java [hunk 4]
```diff
@@ -27,7 +29,7 @@
         this.action = action;
         this.shellId = shellId;
         this.resourceURI = resourceURI;
-        this.optionSet = optionSet;
+        this.optionSet = optionSet != null ? Collections.unmodifiableList(new ArrayList<>(optionSet)) : Collections.emptyList();
```
Comment (MRamonLeon) [inline src/main/java/hudson/plugins/ec2/win/winrm/soap/Header.java:32]: 👏🏽

**Review by MRamonLeon — APPROVED** [review]

**Review by alecharp — APPROVED** [review]

**Review by res0nance — APPROVED** [review]
I agree with Ramon that dropping immutable in all these cases do not matter, so LGTM

**Comment by rsandell** [comment]
@res0nance Thanks for merging!
When can we get a release?

**Comment by res0nance** [comment]
> @res0nance Thanks for merging!
> When can we get a release?

Hey @rsandell, I'm good to release whenever but #602 seems to have received a comment, might be worth checking it out.

I don't think there is too much urgency around Guava removal but yes we definitely want users to start adopting this version before core(hopefully) manages to remove it.

**Comment by alecharp** [comment]
@res0nance the comment on #602 is ok, you can go ahead and release the plugin.

**Pull request merged**